### PR TITLE
updated English version examples in docs/ of master branch

### PR DESCRIPTION
**Pull request by xumeng723**
5 examples were changed as a rule of  :
overview + goals + experiment(by step or by feature)
specially , added two detailed tables in Mnist examples (contains many nni features)

### CODE REVIEW

File: docs/en_US/mnist_examples.md [hunk 1]
```diff
@@ -1,28 +1,126 @@
...
...mnist)<br>[MNIST in keras](#mnist-keras)<br>[MNIST -- tuning with batch tuner](#mnist-batch)<br>[MNIST -- tuning with hyperband](#mnist-hyperband)<br>[MNIST -- tuning within a nested search space](#mnist-nested)<br>[distributed MNIST (tensorflow) using kubeflow](#mnist-kubeflow-tf)<br>[distributed MNIST (pytorch) using kubeflow](#mnist-kubeflow-pytorch)|[examples/trials/mnist/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist)<br>[examples/trials/mnist-keras/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-keras)<br>[/examples/trials/mnist-batch-tune-keras](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-batch-tune-keras)<br>[examples/trials/mnist-hyperband/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-hyperband)<br>[examples/trials/mnist-cascading-search-space/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-cascading-search-space)<br>[examples/trials/mnist-distributed/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-distributed)<br>[examples/trials/mnist-distributed-pytorch/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-distributed-pytorch)
```
Comment (PurityFan) [inline docs/en_US/mnist_examples.md]:
```suggestion
|api|[MNIST with NNI API](#mnist)<br>[MNIST in keras](#mnist-keras)<br>[MNIST -- tuning with batch tuner](#mnist-batch)<br>[MNIST -- tuning with hyperband](#mnist-hyperband)<br>[MNIST -- tuning within a nested search space](#mnist-nested)<br>[distributed MNIST (tensorflow) using kubeflow](#mnist-kubeflow-tf)<br>[distributed MNIST (pytorch) using kubeflow](#mnist-kubeflow-pytorch)|[examples/trials/mnist/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist)<br>[examples/trials/mnist-keras/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-keras)<br>[examples/trials/mnist-batch-tune-keras](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-batch-tune-keras)<br>[examples/trials/mnist-hyperband/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-hyperband)<br>[examples/trials/mnist-cascading-search-space/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-cascading-search-space)<br>[examples/trials/mnist-distributed/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-distributed)<br>[examples/trials/mnist-distributed-pytorch/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-distributed-pytorch)
```

Comment (PurityFan) [reply docs/en_US/mnist_examples.md]: extra **'\'** in '/examples/trials/mnist-batch-tune-keras'

File: docs/en_US/cifar10_examples.md [hunk 2]
```diff
@@ -65,20 +63,20 @@ code directory: [examples/trials/cifar10_pytorch/config_pai.yml][7]
 
 *The complete examples we have implemented: [examples/trials/cifar10_pytorch/][1]*
 
-#### Launch the experiment
+### Lauch the experiment
```
Comment (PurityFan) [inline docs/en_US/cifar10_examples.md]:
```suggestion
### Launch the experiment
```

Comment (PurityFan) [reply docs/en_US/cifar10_examples.md]: Why you change this? I don't think it's a typo.

File: docs/en_US/cifar10_examples.md [hunk 3]
```diff
@@ -41,19 +39,19 @@ As we stated in the target, we target to find out the best `optimizer` for train
...
 
-**Trial**
+### Trial
 
 The code for CNN training of each hyperparameters set, paying particular attention to the following points are specific for NNI:
 
 * Use `nni.get_next_parameter()` to get next training hyperparameter set.
 * Use `nni.report_intermediate_result(acc)` to report the intermedian result after finish each epoch.
-* Use `nni.report_final_result(acc)` to report the final result before the trial end.
+* Use `nni.report_intermediate_result(acc)` to report the final result before the trial end.
```
Comment (PurityFan) [inline docs/en_US/cifar10_examples.md]: Why you change this?

File: docs/en_US/cifar10_examples.md [hunk 4]
```diff
@@ -65,20 +63,20 @@ code directory: [examples/trials/cifar10_pytorch/config_pai.yml][7]
...
+ ```bash
+    nnictl create --config nni/examples/trials/cifar10_pytorch/config.yml
 ```
 
 [1]: https://github.com/Microsoft/nni/tree/master/examples/trials/cifar10_pytorch
 [2]: https://pytorch.org/
 [3]: https://www.cs.toronto.edu/~kriz/cifar.html
 [4]: https://github.com/Microsoft/nni/tree/master/examples/trials/cifar10_pytorch
-[5]: Trials.md
+[5]: https://github.com/Microsoft/nni/blob/master/docs/howto_1_WriteTrial.md
```
Comment (PurityFan) [inline docs/en_US/cifar10_examples.md]: This is a dead link, please use the original one

File: docs/en_US/SQuAD_evolution_examples.md [hunk 5]
```diff
@@ -111,18 +106,18 @@ trial:
   gpuNum: 0
   cpuNum: 1
   memoryMB: 32869
-  #The docker image to run nni job on OpenPAI
+  #The docker image to run nni job on pai
```
Comment (xuehui1991) [inline docs/en_US/SQuAD_evolution_examples.md]: why not change "OpenPAI" intead of "pai"?

File: docs/en_US/cifar10_examples.md [hunk 6]
```diff
@@ -4,17 +4,17 @@
 
 [CIFAR-10][3] classification is a common benchmark problem in machine learning. The CIFAR-10 dataset is the collection of images. It is one of the most widely used datasets for machine learning research which contains 60,000 32x32 color images in 10 different classes. Thus, we use CIFAR-10 classification as an example to introduce NNI usage.
 
-### **Goals**
+## Goals
 
-As we all know, the choice of model optimizer is directly affects the performance of the final metrics. The goal of this tutorial is to **tune a better performace optimizer** to train a relatively small convolutional neural network (CNN) for recognizing images.
+As we all know, the choice of model optimizer is directly affects the performance of the final matrix. The goal of this tutorial is to **tune a better performace optimizer** to train a relatively small convolutional neural network (CNN) for recognizing images.
```
Comment (xuehui1991) [inline docs/en_US/cifar10_examples.md]: "metrics" here, not "matrix"

File: docs/en_US/SQuAD_evolution_examples.md [hunk 7]
```diff
@@ -1,31 +1,17 @@
 # Automatic Model Architecture Search for Reading Comprehension
```
Comment (scarlett2018) [inline docs/en_US/SQuAD_evolution_examples.md]: This comment is not to this single md, but to the entire PR.

may we add a ReadMe.md to the nni/examples folder, so that users who directly go to that root can figure out the examples in a glance

File: docs/en_US/SQuAD_evolution_examples.md [hunk 8]
```diff
@@ -1,31 +1,17 @@
...
-
-![](../../examples/trials/ga_squad/ga_squad.png)
 
-### New version
-Also we have another version which time cost is less and performance is better. We will release soon.
+## Overview
+This example shows us how to use Genetic Algorithm to find good model architectures for Reading Comprehension.
 
-## 2. How to run this example in local?
+## Experimental
```
Comment (scarlett2018) [inline docs/en_US/SQuAD_evolution_examples.md]: NNI use "Experiment" instead of "Experimental" in most of the docs. Please confirm with others if you are on purpose to refactor this wording. Otherwise, please use the common used term.

File: docs/en_US/SQuAD_evolution_examples.md [hunk 9]
```diff
@@ -1,31 +1,17 @@
...
-7. REMOVE-SKIP (Removes random skip).
-
-![](../../examples/trials/ga_squad/ga_squad.png)
 
-### New version
-Also we have another version which time cost is less and performance is better. We will release soon.
+## Overview
+This example shows us how to use Genetic Algorithm to find good model architectures for Reading Comprehension.
 
-## 2. How to run this example in local?
```
Comment (scarlett2018) [inline docs/en_US/SQuAD_evolution_examples.md]: do you want to merge this to overview?

File: docs/en_US/SQuAD_evolution_examples.md [hunk 10]
```diff
@@ -1,31 +1,17 @@
...
-
-1. IDENTITY (Effectively means keep training).
-2. INSERT-RNN-LAYER (Inserts a LSTM. Comparing the performance of GRU and LSTM in our experiment, we decided to use LSTM here.)
-3. REMOVE-RNN-LAYER
-4. INSERT-ATTENTION-LAYER(Inserts an attention layer.)
-5. REMOVE-ATTENTION-LAYER
-6. ADD-SKIP (Identity between random layers).
-7. REMOVE-SKIP (Removes random skip).
-
-![](../../examples/trials/ga_squad/ga_squad.png)
```
Comment (scarlett2018) [inline docs/en_US/SQuAD_evolution_examples.md]: this image seems lost in your new changes

File: docs/en_US/SQuAD_evolution_examples.md [hunk 11]
```diff
@@ -34,22 +20,41 @@ Or Download manually
...
+3. REMOVE-RNN-LAYER
+4. INSERT-ATTENTION-LAYER(Inserts an attention layer.)
+5. REMOVE-ATTENTION-LAYER
+6. ADD-SKIP (Identity between random layers).
+7. REMOVE-SKIP (Removes random skip).
+
+![](../examples/trials/ga_squad/ga_squad.png)
+
+### 3.New version
+
```
Comment (scarlett2018) [inline docs/en_US/SQuAD_evolution_examples.md]: can you confirm what's the corresponding Github Issue for this upcoming "new version"? and add it here.

File: docs/en_US/SQuAD_evolution_examples.md [hunk 12]
```diff
@@ -73,21 +78,11 @@ trial:
...
 
-### 2.3 submit this job
-
-```bash
-nnictl create --config ~/nni/examples/trials/ga_squad/config.yml
-```
-
-## 3 Run this example on OpenPAI
-
+On pai:
```
Comment (scarlett2018) [inline docs/en_US/SQuAD_evolution_examples.md]: this need a refactor, please find me offline.

File: docs/en_US/SQuAD_evolution_examples.md [hunk 13]
```diff
@@ -34,22 +20,41 @@ Or Download manually
 
 1. download "dev-v1.1.json" and "train-v1.1.json" in https://rajpurkar.github.io/SQuAD-explorer/
 
-```bash
+```
```
Comment (xuehui1991) [inline docs/en_US/SQuAD_evolution_examples.md]: why remove bash here?

File: docs/en_US/SQuAD_evolution_examples.md [hunk 14]
```diff
@@ -34,22 +20,41 @@ Or Download manually
...
+### 3.New version
+
+Also we have another version which time cost is less and performance is better. We will release soon.
+
+### 4.Update configuration
+On local:
 Modify `nni/examples/trials/ga_squad/config.yml`, here is the default configuration:
 
-```yaml
+```
```
Comment (xuehui1991) [inline docs/en_US/SQuAD_evolution_examples.md]: why remove "yaml"???

File: docs/en_US/SQuAD_evolution_examples.md [hunk 15]
```diff
@@ -159,7 +157,7 @@ Among those files, `trial.py` and `graph_to_tf.py` are special.
 
 `graph_to_tf.py` has a function named as `graph_to_network`, here is its skeleton code:
 
-```python
+```
```
Comment (xuehui1991) [inline docs/en_US/SQuAD_evolution_examples.md]: why remove "python"????

File: docs/en_US/mnist_examples.md [hunk 16]
```diff
@@ -1,28 +1,126 @@
...
+|pytorch|[distributed MNIST (pytorch) using kubeflow](#mnist-kubeflow-pytorch)|[examples/trials/mnist-distributed-pytorch/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-distributed-pytorch)|
+|keras|[MNIST in keras](#mnist-keras)<br>[MNIST -- tuning with batch tuner](#mnist-batch)|[examples/trials/mnist-keras/](https://github.com/Microsoft/nni/tree/master/examples/trials/mnist-keras)<br>
+
+
+## Experimental
```
Comment (xuehui1991) [inline docs/en_US/mnist_examples.md]: Experimental???

File: docs/en_US/sklearn_examples.md [hunk 17]
```diff
@@ -1,61 +1,46 @@
...
+  ```
   {
     "model_name":{"_type":"choice","_value":["LinearRegression", "SVR", "KNeighborsRegressor", "DecisionTreeRegressor"]}
   }
   ```
-
   If you want to choose different models and parameters, you could put them together in a search_space.json file.
-
-  ```json
+  ```
```
Comment (xuehui1991) [inline docs/en_US/sklearn_examples.md]: same here

File: docs/en_US/sklearn_examples.md [hunk 18]
```diff
@@ -1,61 +1,46 @@
...
+* __step 1__  
 
-* __step 1__
-
-  Prepare a search_space.json to storage your choose spaces.
+  Prepare a search_space.json to storage your choose spaces. 
   For example, if you want to choose different models, you may try:
-
-  ```json
+  ```
```
Comment (xuehui1991) [inline docs/en_US/sklearn_examples.md]: same here.

File: docs/en_US/sklearn_examples.md [hunk 19]
```diff
@@ -1,61 +1,46 @@
...
   ```
-
   Then you could read these values as a dict from your python code, please get into the step 2.
 * __step 2__  
   At the beginning of your python code, you should `import nni` to insure the packages works normally.
   First, you should use `nni.get_next_parameter()` function to get your parameters given by nni. Then you could use these parameters to update your code.
   For example, if you define your search_space.json like following format:
-
-  ```json
+  ```
```
Comment (xuehui1991) [inline docs/en_US/sklearn_examples.md]: same here.

File: docs/en_US/cifar10_examples.md [hunk 20]
```diff
@@ -65,20 +63,20 @@ code directory: [examples/trials/cifar10_pytorch/config_pai.yml][7]
...
+ ```bash
+    nnictl create --config nni/examples/trials/cifar10_pytorch/config.yml
 ```
 
 [1]: https://github.com/Microsoft/nni/tree/master/examples/trials/cifar10_pytorch
 [2]: https://pytorch.org/
 [3]: https://www.cs.toronto.edu/~kriz/cifar.html
 [4]: https://github.com/Microsoft/nni/tree/master/examples/trials/cifar10_pytorch
-[5]: Trials.md
+[5]: https://github.com/Microsoft/nni/blob/master/docs/howto_1_WriteTrial.md
```
Comment (xuehui1991) [inline docs/en_US/cifar10_examples.md]: ? 404?

File: docs/en_US/mnist_examples.md [hunk 21]
```diff
@@ -1,28 +1,126 @@
...
+
+
+
+<a name ="mnist"></a>
 **MNIST with NNI API**
 
-This is a simple network which has two convolutional layers, two pooling layers and a fully connected layer. We tune hyperparameters, such as dropout rate, convolution size, hidden size, etc. It can be tuned with most NNI built-in tuners, such as TPE, SMAC, Random. We also provide an exmaple YAML file which enables assessor.
+This is a simple network which has two convolutional layers, two pooling layers and a fully connected layer. We tune hyperparameters, such as dropout rate, convolution size, hidden size, etc. It can be tuned with most NNI built-in tuners, such as TPE, SMAC, Random. We also provide an exmaple yaml file which enables assessor.
+
+As we stated in the target, we target to find out the best `optimizer` for training . When using different optimizers, we also need to adjust `dropout_rate` ,`conv_size`,`hidden_size`,`batch_size`and `learning_rate` accordingly. so we chose these three parameters as hyperparameters and write the following search space.
```
Comment (xuehui1991) [inline docs/en_US/mnist_examples.md]: I think " we target to find out the best `optimizer` for training " belong to the "pytorch example".... I don't know why you add this sentence here...

File: docs/en_US/mnist_examples.md [hunk 22]
```diff
@@ -35,15 +133,27 @@ This example is implemented in keras. It is also a network for MNIST dataset, wi
 <a name="mnist-batch"></a>
 **MNIST -- tuning with batch tuner**
 
-This example is to show how to use batch tuner. Users simply list all the configurations they want to try in the search space file. NNI will try all of them.
-
+This example is to show how to use batch tuner. Users simply list all the configurations they want to try in the search space file. NNI will try all of them.If the configurations you want to try have been decided, you can list them in searchspace file (using `choice`) and run them using batch tuner.
+```yaml
```
Comment (xuehui1991) [inline docs/en_US/mnist_examples.md]: why remove the yaml here?
